### PR TITLE
add cmake script for damaris handling

### PR DESCRIPTION
**Pull request by akva2**
Upstream of https://github.com/OPM/opm-simulators/pull/4121

### CODE REVIEW

**Comment by blattms** [comment]
just one question. Upstream seems to use CMake. Don't they ship any CMake configuration files (damaris-config.cmake or DamarisConfig.cmake)? If not, then they should. Maybe ask them to do this.

**Comment by akva2** [comment]
they do. this is just verbose printing + config.h variable handling.

File: cmake/Modules/UseDamaris.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      message(STATUS "The Damaris library does NOT have Catalyst support")
+    endif()
+  else()
+    message(STATUS "The Damaris library was requested but NOT found")
+  endif()
+else()  # User did not request Damaris support
+  if (Damaris_FOUND)
+    message(STATUS "The Damaris library was NOT requested but was found: ${Damaris_VERSION} ${Damaris_DIR}")
+  else()
+    message(STATUS "The Damaris library was NOT requested and was NOT found")
```
Comment (bska) [inline cmake/Modules/UseDamaris.cmake:27]: Do we really need/want this particular diagnostic message?  Why does the user need to know anything about Damaris if they've not asked for it?

Comment (akva2) [reply cmake/Modules/UseDamaris.cmake:27]: in my opinion, no, but it was there in the PR I refactored.

Comment (bska) [reply cmake/Modules/UseDamaris.cmake:27]: >> Do we really need/want this particular diagnostic message?
>
> in my opinion, no, but it was there in the PR I refactored.

Understood.  Maybe we could trim the diagnostics down a bit then?  I don't really care if we do that before or after merging this PR.

File: cmake/Modules/UseDamaris.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  endif()
+  unset(HAVE_DAMARIS)
+endif()
+
+mark_as_advanced(HAVE_DAMARIS)
```
Comment (atgeirr) [inline cmake/Modules/UseDamaris.cmake:32]: Naive question here: Is it not the option name (USE_DAMARIS_LIB) that should be marked as advanced, not the name of the define for the config? (Assuming "advanced" here refers to the toggle in ccmake or equivalent.)

Comment (akva2) [reply cmake/Modules/UseDamaris.cmake:32]: advanced means it's hidden in ccmake.

**Comment by atgeirr** [comment]
I assume this can be merged standalone?

**Comment by akva2** [comment]
yes this in itself is a NOP, it just adds a (n unused) file.

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by ElyesAhmed** [comment]
>


> just one question. Upstream seems to use CMake. Don't they ship any CMake configuration files (damaris-config.cmake or DamarisConfig.cmake)? If not, then they should. Maybe ask them to do this.

They do, there is damaris-config-version.cmake

**Comment by ElyesAhmed** [comment]
> just one question. Upstream seems to use CMake. Don't they ship any CMake configuration files (damaris-config.cmake or DamarisConfig.cmake)? If not, then they should. Maybe ask them to do this.

Yes they do.

**Comment by atgeirr** [comment]
This is green (no surprise), merging now and avoiding cross-repo jenkins invocation downstream.

**Pull request merged**